compiler/ccgtypes.nim

```diff
@@ -1087,6 +1094,8 @@ proc getTypeDescAux(m: BModule; origTyp: PType, check: var IntSet; kind: TypeDes
     result = ""
   # fixes bug #145:
   excl(check, t.id)
+  if kind == dkRefGenericParam:
+    result.add "&"
```

Comment:
Why not do that for every `dkRefParam` here?

Reply:
That point is unreachable, sorry. It was the result of a previous attempt for fixing the forward declaration. Will remove it and push back, once you are ok with the one below.